src/Countries/Malaysia.php

```diff
@@ -0,0 +1,540 @@
+            'Hari Thaipusam' => $this->hariThaipusam($year),
+            'Hari Nurul Al-Quran' => $this->nuzulQuran($year),
+            'Hari Deepavali' => $this->hariDeepavali($year),
+            'Hari Keputeraan Sultan Selangor' => CarbonImmutable::createFromDate($year, 12, 11, $this->timezone),
+        ];
+    }
+
+    /**
+     * @return array<string, CarbonImmutable|string>
+     */
+    protected function regionTerengganu(int $year): array
+    {
+        return [
+            'Israk dan Mikraj' => $this->israkMikraj($year),
+            'Hari Ulang Tahun Pertabalan Sultan Terengganu' => CarbonImmutable::createFromDate($year, 3, 4, $this->timezone),
+            'Hari Nurul Al-Quran' => $this->nuzulQuran($year),
+            'Hari Keputeraan Sultan Terengganu' => CarbonImmutable::createFromDate($year, 4, 26, $this->timezone),
+            'Hari Arafah' => $this->hariArafah($year),
+            'Hari Raya Aidiladha Hari Kedua' => $this->hariAidiladha($year)->addDays(1),
+            'Hari Deepavali' => $this->hariDeepavali($year),
+        ];
+    }
+
+    private function newYear(): string
+    {
+        return '01-01';
+    }
+
+    private function lunarNewYear(int $year): CarbonImmutable
+    {
+        return $this->chineseToGregorianDate('01-01', $year);
+    }
+
+    protected function getIslamicFormatter(): IntlDateFormatter
+    {
+        return new IntlDateFormatter(
+            locale: 'ms_MY@calendar=islamic-civil',
+            dateType: IntlDateFormatter::MEDIUM,
+            timeType: IntlDateFormatter::NONE,
+            timezone: $this->timezone,
+            calendar: IntlDateFormatter::TRADITIONAL
+        );
+    }
+
+    protected function getHijriYear(int $year, bool $nextYear = false): int
+    {
+        $formatter = $this->getIslamicFormatter();
+        $formatter->setPattern('yyyy');
+        $dateTime = DateTime::createFromFormat('d/m/Y', '01/01/' . ($nextYear ? $year + 1 : $year));
+
+        if (!$dateTime) {
+            throw InvalidYear::invalidHijriYear();
+        }
+
+        return (int) $formatter->format($dateTime->getTimestamp());
+    }
+
+    /**
+     * $input as in 'day-month'; 01-10 for syawal
+     */
+    protected function islamicCalendar(string $input, int $year, bool $nextYear = false): CarbonImmutable
+    {
+        $hijrYear = $this->getHijriYear(year: $year, nextYear: $nextYear);
+        $formatter = $this->getIslamicFormatter();
+        $timeStamp = (int) $formatter->parse($input . '/' . $hijrYear . ' AH');
+        
+        return CarbonImmutable::createFromTimestamp($timeStamp, $this->timezone);
+    }
+
+    protected function hariAidilfitri(int $year): CarbonImmutable
+    {
+        return $this->islamicCalendar('01/10', $year);
+    }
+
+    protected function hariWilayah(int $year): CarbonImmutable
+    {
+        return CarbonImmutable::createFromDate($year, 2, 1, $this->timezone);
+    }
+
+    protected function israkMikraj(int $year): CarbonImmutable
```

Comment:
I don't really like all this separate functions. You can just use `$this->islamicCalendar();` instead